components/blitz/src/ome/services/blitz/repo/ManagedImportProcessI.java

```diff
@@ -208,43 +227,58 @@ public void shutdown() {
     public RawFileStorePrx getUploader(final int i, Current current)
             throws ServerError {
 
-        String mode = null;
-        if (current != null && current.ctx != null) {
-            mode = current.ctx.get("omero.fs.mode");
-            if (mode == null) {
-                mode = "rw";
+        StopWatch sw1 = new Slf4JStopWatch();
+        try {
+            MDC.put("fileset", logFilename);
+
+            String mode = null;
+            if (current != null && current.ctx != null) {
+                mode = current.ctx.get("omero.fs.mode");
+                if (mode == null) {
+                    mode = "rw";
+                }
             }
-        }
 
-        final String applicableMode = mode;
-        final Callable<UploadState> rfsOpener = new Callable<UploadState>() {
-            @Override
-            public UploadState call() throws ServerError {
-                final String path = location.sharedPath + FsFile.separatorChar + location.usedFiles.get(i);
-                final RawFileStorePrx prx = repo.file(path, applicableMode, ManagedImportProcessI.this.current);
-                try {
-                    registerCallback(prx, i);
-                } catch (RuntimeException re) {
+            final String applicableMode = mode;
+            final Callable<UploadState> rfsOpener = new Callable<UploadState>() {
+                @Override
+                public UploadState call() throws ServerError {
+                    final StopWatch sw2 = new Slf4JStopWatch();
+                    final String path = location.sharedPath + FsFile.separatorChar + location.usedFiles.get(i);
+                    final Ice.Current adjustedCurr = repo.makeAdjustedCurrent(ManagedImportProcessI.this.current);
+                    adjustedCurr.ctx.put(CallContext.FILENAME_KEY, logFilename);
+                    adjustedCurr.ctx.put(CallContext.TOKEN_KEY, rootToken);
+                    final RawFileStorePrx prx = repo.file(path, applicableMode,  adjustedCurr);
                     try {
-                        prx.close();  // close if anything happens
-                    } catch (Exception e) {
-                        log.error("Failed to close RawFileStorePrx", e);
+                        registerCallback(prx, i);
+                    } catch (RuntimeException re) {
+                        try {
+                            prx.close();  // close if anything happens
+                        } catch (Exception e) {
+                            log.error("Failed to close RawFileStorePrx", e);
+                        }
+                        throw re;
+                    } finally {
+                        sw2.stop("omero.import.process.opener");
+                        MDC.clear();
```

Comment:
I don't know how the `MDC.clear()` here interacts with the one below outside this `Callable`. Both "opener" and "uploader" do seem to get time in the log.

Reply:
Sure. We can revisit if need be.